cmd/commands/datasetIngestor.go

```diff
@@ -259,123 +267,133 @@ For Windows you need instead to specify -user username:password on the command l
 				log.Fatalf("Can't gather the filelist of \"%s\"", datasetSourceFolder)
 			}
+
 			//log.Printf("full fileListing: %v\n Start and end time: %s %s\n ", fullFileArray, startTime, endTime)
 			log.Printf("The dataset contains %v files with a total size of %v bytes.", numFiles, totalSize)
 
+			// filecount checks
 			if totalSize == 0 {
 				emptyDatasets++
 				color.Set(color.FgRed)
-				log.Println("This dataset contains no files and will therefore NOT be stored. ")
+				log.Printf("\"%s\" dataset cannot be ingested - contains no files\n", datasetSourceFolder)
 				color.Unset()
-			} else if numFiles > TOTAL_MAXFILES {
+				continue
+			}
+			if numFiles > TOTAL_MAXFILES {
 				tooLargeDatasets++
 				color.Set(color.FgRed)
-				log.Printf("This dataset exceeds the current filecount limit of the archive system of %v files and will therefore NOT be stored.\n", TOTAL_MAXFILES)
+				log.Printf("\"%s\" dataset cannot be ingested - too many files: has %d, max. %d\n", datasetSourceFolder, numFiles, TOTAL_MAXFILES)
 				color.Unset()
-			} else {
-				// TODO: change tapecopies param type of UpadateMetaData from pointer to regular int
-				// (it's not changed within the function)
-				datasetIngestor.UpdateMetaData(client, APIServer, user, originalMap, metaDataMap, startTime, endTime, owner, tapecopies)
-				pretty, _ := json.MarshalIndent(metaDataMap, "", "    ")
-
-				log.Printf("Updated metadata object:\n%s\n", pretty)
-
-				// check if data is accesible at archive server, unless beamline account (assumed to be centrally available always)
-				// and unless copy flag defined via command line
-				if !copyFlag && !nocopyFlag { // NOTE this whole copyFlag, nocopyFlag ordeal makes no sense whatsoever
-					if !beamlineAccount {
-						sshErr, otherErr := datasetIngestor.CheckDataCentrallyAvailableSsh(user["username"], RSYNCServer, datasetSourceFolder, os.Stdout)
-						if otherErr != nil {
-							log.Fatalf("CheckDataCentrallyAvailableSsh returned an error: %v\n", otherErr)
-						}
-						if sshErr != nil {
-							color.Set(color.FgYellow)
-							log.Printf("The source folder %v is not centrally available (decentral use case).\nThe data must first be copied to a rsync cache server.\n ", datasetSourceFolder)
-							color.Unset()
-							copyFlag = true
-							// check if user account
-							if len(accessGroups) == 0 {
-								color.Set(color.FgRed)
-								log.Println("For the decentral case you must use a personal account. Beamline accounts are not supported.")
-								color.Unset()
-								os.Exit(1)
-							}
-							if !noninteractiveFlag {
-								log.Printf("Do you want to continue (Y/n)? ")
-								scanner.Scan()
-								continueFlag := scanner.Text()
-								if continueFlag == "n" {
-									log.Fatalln("Further ingests interrupted because decentral case detected, but no copy wanted.")
-								}
-							}
+				continue
+			}
+
+			// NOTE: only tapecopies=1 or 2 does something if set.
+			if tapecopies == 2 {
```

Comment:
I'm not sure. The underlying code in datasetIngestor/updateMetaData.go, specifically updateClassificationField function, only updates the corresponding field for the metadata if the value is either 1 or 2, nothing else. It could be on purpose, if, for example, we don't want people to create more than 2 copies.

Reply:
In any case, the code can't do more than two copies, so I'm fine leaving it as is for now